strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactController.java

```diff
@@ -277,11 +281,22 @@ public void viewPackageFeedWithScope(@RepositoryMapping Repository repository,
             }
 
         });
-
+        packageFeed.setAdditionalProperty("_rev", generateRevisionHashcode(packageFeed));
         response.setContentType(MediaType.APPLICATION_JSON_VALUE);
         response.getOutputStream().write(npmJacksonMapper.writeValueAsBytes(packageFeed));
     }
 
+    private String generateRevisionHashcode(PackageFeed packageFeed)
+    {
+        String versionsShasum = packageFeed.getVersions().getAdditionalProperties()
+                                           .values()
+                                           .stream()
+                                           .map(x -> x.getDist().getShasum())
+                                           .collect(Collectors.joining());
+        return packageFeed.getVersions().getAdditionalProperties().size() + "-"
+               + DigestUtils.sha1Hex(versionsShasum).substring(0, 16);
```

Comment:
Cosmetic: We usually put the `+` at the end of the line when it's a multi-line statement like this:
```java
        return packageFeed.getVersions().getAdditionalProperties().size() + "-" +
               DigestUtils.sha1Hex(versionsShasum).substring(0, 16);
```